integration-tests/src/test/java/gov/uk/kbv/api/stepdefinitions/CertificateExpiryStepDefinition.java

```diff
@@ -0,0 +1,219 @@
+                                "SHA256": "AA:AA:AA:AA:AA:AA:AA:AA:AA:AA:A****************************************************************",
+                                "SHA1": "AA:AA:AA:AA:AA:AA:A****************************************"
+                              }
+                            }
+                          ],
+                          "creationDate": "Jan 01, 2025"
+                        }
+                      ]
+                    }
+                    """;
+
+    private String stateMachineInput;
+    private String executionArn;
+
+    @Given("no certificate overrides are passed into the step function")
+    public void noCertificateOverridesArePassedIntoTheStepFunction() {
+        stateMachineInput = "{}";
+    }
+
+    @Given("no certificates are expiring soon")
+    public void noCertificatesAreExpiringSoon() {
+        stateMachineInput = String.format(DUMMY_CERT_JSON, NOW.plusYears(1).format(FORMATTER));
+    }
+
+    @Given("a certificate is expiring within 90 days")
+    public void aCertificateIsExpiringWithin90Days() {
+        stateMachineInput = String.format(DUMMY_CERT_JSON, NOW.plusDays(50).format(FORMATTER));
+    }
+
+    @Given("a certificate is expiring within 7 days")
+    public void aCertificateIsExpiringWithin7Days() {
+        stateMachineInput = String.format(DUMMY_CERT_JSON, NOW.plusDays(1).format(FORMATTER));
+    }
+
+    @Given("three certificates with various expiries are passed into the step function")
+    public void threeCertificatesArePassedIntoTheStepFunction() {
+        stateMachineInput =
+                String.format(
+                        THREE_CERTS_JSON,
+                        NOW.plusYears(1).format(FORMATTER),
+                        NOW.plusDays(80).format(FORMATTER),
+                        NOW.plusDays(5).format(FORMATTER));
+    }
+
+    @When("the step function is invoked")
+    public void theStepFunctionIsInvoked() {
+        String certificateExpiryStepFunctionArn =
+                CloudFormationHelper.getOutput(
+                        System.getenv("STACK_NAME"), "CertificateExpiryStepFunctionArn");
+
+        StartExecutionResponse response =
+                SFN_CLIENT.startExecution(
+                        StartExecutionRequest.builder()
+                                .stateMachineArn(certificateExpiryStepFunctionArn)
+                                .name(
+                                        String.join(
+                                                "-",
+                                                "integration-test",
+                                                UUID.randomUUID().toString()))
+                                .input(stateMachineInput)
+                                .build());
+        executionArn = response.executionArn();
+    }
+
+    @SuppressWarnings("java:S2925")
+    private ExecutionStatus pollForExecutionStatus() throws InterruptedException {
+        for (int count = 0; count < 20; count++) {
```

Comment:
Just wondering if we could have used Retry thing here -> https://github.com/govuk-one-login/ipv-cri-lib/blob/main/src/main/java/uk/gov/di/ipv/cri/common/library/service/SessionService.java#L154 or maybe that overkill